### PR TITLE
Make styleguidist server publicly accessible

### PR DESCRIPTION
**Pull request by leclercb**
Add doc: "How to make styleguidist server publicly accessible?"

### CODE REVIEW

File: docs/Cookbook.md [hunk 1]
```diff
@@ -254,6 +255,21 @@ module.exports = {
 
 ![](https://d3vv6lp55qjaqc.cloudfront.net/items/3i3E3j2h3t1315141k0o/debugging.png)
 
+## How to make styleguidist server publicly accessible?
+
+In order to make styleguidist server publicly accessible, add the following to the styleguide.config.js file:
```
Comment (sapegin) [inline docs/Cookbook.md]: * styleguidist → Styleguidist

File: docs/Cookbook.md [hunk 2]
```diff
@@ -257,6 +257,8 @@ module.exports = {
 
 ## How to make styleguidist server publicly accessible?
 
+Warning: this setup is not recommended for security reasons, it should only be enabled on dev environments.
```
Comment (sapegin) [inline docs/Cookbook.md]: I don’t think the second part is necessary, I think it’s dangerous in any case ;-)

Also: `**Warning:**`

File: docs/Cookbook.md [hunk 3]
```diff
@@ -257,6 +257,8 @@ module.exports = {
 
 ## How to make styleguidist server publicly accessible?
```
Comment (sapegin) [inline docs/Cookbook.md]: styleguidist → Styleguidist

**Comment by n1313** [comment]
It would be a good addition to add a link to the original issue https://github.com/webpack/webpack-dev-server/issues/887 to illustrate what exactly the security risk here is about.

File: docs/Cookbook.md [hunk 4]
```diff
@@ -254,6 +255,23 @@ module.exports = {
...
+
+Warning: this setup is not recommended for security reasons, it should only be enabled on dev environments.
+
+In order to make styleguidist server publicly accessible, add the following to the styleguide.config.js file:
+
+```javascript
+module.exports = {
+  webpackConfig: {
+    ...
+    devServer: {
```
Comment (roblevintennis) [inline docs/Cookbook.md]: Fwiw we just had to do this. We do a check `!production` in our whole dev-server config so it's "safe" I suppose. Yeah, I think this is helpful as I spent a good hour digging in to figure this out myself :) 👍

Comment (roblevintennis) [reply docs/Cookbook.md]: Hrm, actually, we're electing NOT to do this in favor of something more like:
```
const sg = process.env.STYLEGUIDE === 'true';
...
} else if (sg) {
  config.output.publicPath = '//path.local/';

in our main webpack.config.js

**Pull request closed without merging**